pkg/tenant/topics.go

```diff
@@ -317,46 +316,61 @@ func (s *Server) TopicDelete(c *gin.Context) {
 	// Fetch the topic metadata from the database
 	var topic *db.Topic
 	if topic, err = db.RetrieveTopic(ctx, topicID); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			log.Warn().Err(err).Str("topicID", topicID.String()).Msg("topic not found")
+			c.JSON(http.StatusNotFound, api.ErrorResponse("topic not found"))
+			return
+		}
 		log.Error().Err(err).Str("topicID", topicID.String()).Msg("could not retrieve topic")
-		c.JSON(http.StatusNotFound, api.ErrorResponse("topic not found"))
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not delete topic"))
 		return
 	}
 
 	// Verify that the user owns the topic
 	if claims.OrgID != topic.OrgID.String() {
 		log.Warn().Err(err).Str("user_org", claims.OrgID).Str("topic_org", topic.OrgID.String()).Msg("topic OrgID does not match user OrgID")
-		c.JSON(http.StatusForbidden, api.ErrorResponse("user is not authorized to delete this topic"))
+		c.JSON(http.StatusNotFound, api.ErrorResponse("topic not found"))
 		return
 	}
 
 	// Send confirmation token if not provided
-	if confirm.ConfirmToken == "" {
+	if confirm.Token == "" {
 		// Create a short-lived confirmation token in the database
-		topic.ConfirmDeleteToken = secrets.CreateToken(16)
-		topic.ConfirmDeleteExpire = time.Now().Add(5 * time.Minute)
+		if topic.ConfirmDeleteToken, err = db.NewResourceToken(topic.OrgID); err != nil {
```

Comment:
Why is the resource token ID the topic's OrgID? I was expecting it to be the topic ID?